src/Core/src/Platform/iOS/ApplicationExtensions.cs

```diff
@@ -57,14 +58,23 @@ public static void CreatePlatformWindow(this IUIWindowSceneDelegate sceneDelegat
 				dicts.Add(session.UserInfo);
 			if (session.StateRestorationActivity?.UserInfo is not null)
 				dicts.Add(session.StateRestorationActivity.UserInfo);
-			if (connectionOptions.UserActivities is not null)
+			try
 			{
-				foreach (var u in connectionOptions.UserActivities)
+				using var activities = connectionOptions.UserActivities;
+				if (activities is not null)
 				{
-					if (u is NSUserActivity userActivity && userActivity.UserInfo is not null)
-						dicts.Add(userActivity.UserInfo);
+					foreach (var u in activities)
+					{
+						if (u is NSUserActivity userActivity && userActivity.UserInfo is not null)
+							dicts.Add(userActivity.UserInfo);
+					}
 				}
 			}
+			catch (InvalidCastException)
+			{
+				// HACK: Workaround for https://github.com/xamarin/xamarin-macios/issues/13704
+				//       This only throws if the collection is empty.
+			}
```

Comment:
Second-point-five protection, if we do somehow throw, then we know there is no user activity.